test/models/user_test.rb

```diff
@@ -0,0 +1,27 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  
+  def setup
+    foo = ('a'..'z').to_a.shuffle.first(8).join # random 8char string for unique email
+    @user = User.new({:email => "#{foo}@gmail.com", :first_name => "someFirstName", 
+                      :last_name => "someLastName", :password => "111111", 
+                      :password_confirmation => "111111" }).save(:validate => false)
+    # save(:validate) is needed only when you use email validation, I think?
+      #no, actually you've got to have it. Gotta save, and gotta ( => false).
+    # as a side-note, when .save() is used -  foo == true. Not exactly sure why
+    puts @user[:email]
+  end
+  
+  test "user should be valid" do
+    assert @user.valid?
+  end
+  
+  test "first_name has to be proper length" do
+    names = [nil, '', 'a', '123456789012345678901234567890']
+    names.each do |invalid_name|
+      @user[:first_name] = invalid_name # @user.first_name works as well here
+      assert_not @user.valid? "#{invalid_name} shouldn't be valid" # shows which is wrong
+    end
+  end
```

Comment:
@MagHz What are you hoping to test for here?  You already have a character length validation on the `User` model.  Here, you're have a self-contained code block that's not really interacting with the business logic of the app.

An easier way to structure this test might be to simulate the creation of a new User.  Once created, does it have all the properties that you would expect?  If you want full coverage, maybe test invalid inputs like you're doing here but in the context of the entire process.